src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -165,16 +176,25 @@ private static void logClassNotFoundError(String className, Exception ex) {
    * the Java access rules).
    */
   public static Set<Method> getAvailableMethods(Class<?> clazz) {
-    Set<Method> methods = Sets.newHashSet();
-    methods.addAll(Arrays.asList(clazz.getDeclaredMethods()));
+    Map<String, Set<Method>> methods = Maps.newHashMap();
+    for (final Method method : clazz.getDeclaredMethods()) {
+      if (! methods.containsKey(method.getName())) {
```

Comment:
The pattern provided by @cbeust is a bit more optimized because you look for only once.
Here, you look for twice.